src/commands/buildCommands/status.js

```diff
@@ -0,0 +1,119 @@
+const { promisify } = require('util');
+const request = require('request');
+
+// hard-coded to .com, not .org - Travis is migrating everyting to .com
+// but in the interim this script will not work with travis-ci.org builds
+const TRAVIS_API_URL = 'https://api.travis-ci.com';
+
+const _get = promisify(request.get);
+const _post = promisify(request.post);
+const get = (path, options = {}) => _get(`${TRAVIS_API_URL}${path}`, options);
+const post = (path, options = {}) => _post(`${TRAVIS_API_URL}${path}`, options);
+
+// Travis API needs auth token and repo to search for data
+// https://developer.travis-ci.com/
+const getTravisApi = ({ token, repo }) => {
+	const headers = {
+		'Travis-API-Version': 3,
+		Authorization: `token ${token}`,
+	};
+	return {
+		build: {
+			cancel: id =>
+				post(`/build/${id}/cancel`)
+					.then(({ body }) => {
+						console.log(`Canceled build ${id}`);
+					})
+					.catch(console.error),
+			// https://developer.travis-ci.com/resource/build#find
+			get: id =>
+				get(`/build/${id}`, { headers }).then(({ body }) =>
+					JSON.parse(body)
+				),
+			latest: repo =>
+				// 'latest' build of interest are _active_ builds only (started/created)
+				get(
+					`/repo/${encodeURIComponent(
+						repo
+					)}/builds?state=started,created&sort_by=started_at:desc&limit=1`,
+					{ headers }
+				)
+					.then(resp => JSON.parse(resp.body).builds)
+					.then(builds => (builds.length > 0 ? builds[0] : null))
+					.catch(console.error),
+		},
+	};
+};
+
+// make a function that will test if a second build started within the
+// `minInterval` time between builds
+const makeTestShortInterval = (build1, minInterval) => build2 => {
```

Comment:
Assuming this script would be first to run which is why we are only checking for another build?

Reply:
? I'm not sure what you mean. The script will get a 'reference' build ID from the CLI options (`--id` or the `TRAVIS_BUILD_ID` env var), and then this function tests whether any other build has started within `minInterval`)